tests/unit/util/test.inc

```diff
@@ -467,3 +467,62 @@ void run_mixing_GET_ARG_and_FOR_EACH(void)
 	zassert_equal(a[3], 4);
 	zassert_equal(a[4], 5);
 }
+
+void run_IS_ARRAY_ELEMENT(void)
+{
+	size_t i;
+	size_t array[3];
+	uint8_t *const alias = (uint8_t *)array;
+
+	zassert_false(IS_ARRAY_ELEMENT(array, &array[-1]));
+	zassert_false(IS_ARRAY_ELEMENT(array, &array[ARRAY_SIZE(array)]));
+	zassert_false(IS_ARRAY_ELEMENT(array, &alias[1]));
+
+	for (i = 0; i < ARRAY_SIZE(array); ++i) {
+		zassert_true(IS_ARRAY_ELEMENT(array, &array[i]));
+	}
```

Comment:
Minor but general comment: You might as well move the `size_t` declaration inside the loop. 

That way we reduce the scope of `i` which is generally wanted. 

In any case, either way works, and I'll approve.